src/widgets/lv_slider.c

```diff
@@ -330,30 +330,20 @@ static void draw_knob(lv_event_t * e)
     lv_obj_t * obj = lv_event_get_target(e);
     lv_slider_t * slider = (lv_slider_t *)obj;
     lv_draw_ctx_t * draw_ctx = lv_event_get_draw_ctx(e);
-    lv_base_dir_t base_dir = lv_obj_get_style_base_dir(obj, LV_PART_MAIN);
 
     lv_coord_t objw = lv_obj_get_width(obj);
     lv_coord_t objh = lv_obj_get_height(obj);
-    bool hor = objw >= objh ? true : false;
-    lv_coord_t knob_size = hor ? objh : objw;
+    bool is_horizontal = objw >= objh;
     bool sym = false;
     if(slider->bar.mode == LV_BAR_MODE_SYMMETRICAL && slider->bar.min_value < 0 && slider->bar.max_value > 0) sym = true;
 
-    lv_area_t knob_area;
+    lv_base_dir_t base_dir = lv_obj_get_style_base_dir(obj, LV_PART_MAIN);
+    bool is_rtl = base_dir == LV_BASE_DIR_RTL;
 
+    lv_area_t knob_area;
     /*Horizontal*/
-    if(hor) {
-        if(!sym) {
-            knob_area.x1 = LV_SLIDER_KNOB_COORD(hor, base_dir == LV_BASE_DIR_RTL, slider->bar.indic_area);
-        }
-        else {
-            if(slider->bar.cur_value >= 0) {
```

Comment:
Where is the `sym == true` case handled now?

Reply:
This is the previous code:

```c
        if(!sym) {
            knob_area.x1 = LV_SLIDER_KNOB_COORD(hor, base_dir == LV_BASE_DIR_RTL, slider->bar.indic_area);
        }
        else {
            if(slider->bar.cur_value >= 0) {
                knob_area.x1 = LV_SLIDER_KNOB_COORD(hor, base_dir == LV_BASE_DIR_RTL, slider->bar.indic_area);
            }
            else {
                knob_area.x1 = LV_SLIDER_KNOB_COORD(hor, base_dir != LV_BASE_DIR_RTL, slider->bar.indic_area);
            }
        }
```

All the cases is the same assignment, the only thing that changed is the RTL check and it's now using a variable.

Reply:
Ah, you are right! I missed some parts which are actually handles by `lv_bar`.